server/src/ClientToDBAPI/InvestigationInfo/mainRoute.ts

```diff
@@ -129,49 +129,51 @@ investigationInfo.post('/comment', (request: Request, response: Response) => {
+            epidemiologyNumber,
             status: investigationMainStatus
         })}`,
         user: response.locals.user.id,
         investigation: response.locals.epidemiologynumber
     })
     graphqlRequest(UPDATE_INVESTIGATION_STATUS, response.locals, {
-        epidemiologyNumber: parseInt(response.locals.epidemiologynumber),
+        epidemiologyNumber,
         investigationStatus: investigationMainStatus,
         investigationSubStatus: investigationSubStatus
     })
         .then((result: any) => {
-            logger.info({
-                service: Service.SERVER,
-                severity: Severity.LOW,
-                workflow: currentWorkflow,
-                step: 'the investigation status was updated successfully',
-                user: response.locals.user.id,
-                investigation: response.locals.epidemiologynumber
-            });
-            if (investigationMainStatus === InvestigationMainStatus.DONE) {
+            if (result?.data && !result.errors) {
                 logger.info({
-                    service: Service.CLIENT,
+                    service: Service.SERVER,
                     severity: Severity.LOW,
-                    workflow: 'Ending Investigation',
-                    step: `launching graphql API request to update end time with parameters: ${JSON.stringify({
-                        epidemiologyNumber: response.locals.epidemiologynumber,
-                        investigationEndTime: new Date()
-                    })}`,
+                    workflow: currentWorkflow,
+                    step: 'the investigation status was updated successfully',
                     user: response.locals.user.id,
                     investigation: response.locals.epidemiologynumber
                 });
-                graphqlRequest(UPDATE_INVESTIGATION_END_TIME, response.locals, {
-                    epidemiologyNumber: parseInt(response.locals.epidemiologynumber),
-                    investigationEndTime: new Date()
-                })
+                if (investigationMainStatus === InvestigationMainStatus.DONE) {
+                    const investigationEndTime = new Date();
+                    logger.info({
+                        service: Service.CLIENT,
+                        severity: Severity.LOW,
+                        workflow: 'Ending Investigation',
+                        step: `launching graphql API request to update end time with parameters: ${JSON.stringify({
+                            epidemiologyNumber: response.locals.epidemiologynumber,
+                            investigationEndTime
+                        })}`,
+                        user: response.locals.user.id,
+                        investigation: response.locals.epidemiologynumber
+                    });
+                    graphqlRequest(UPDATE_INVESTIGATION_END_TIME, response.locals, {
+                        epidemiologyNumber: parseInt(response.locals.epidemiologynumber),
```

Comment:
maybe use the + syntax instead of parseInt (just to keep consistency)

Reply:
I accidently didnt change it to use the variable props then I did it now